### PR TITLE
ROC-3252 Add custom error decoder

### PR DESCRIPTION
**Pull request by dharmendrak**

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sendletter/CustomFeignErrorDecoder.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.client.RestClientException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+@Slf4j
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/reform/sendletter/CustomFeignErrorDecoder.java]: Adding lombok just for this is over kill, please remove

File: src/main/java/uk/gov/hmcts/reform/sendletter/CustomFeignErrorDecoder.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+public class CustomFeignErrorDecoder implements ErrorDecoder {
+
+    private ErrorDecoder delegate = new ErrorDecoder.Default();
+
+    @Override
+    public Exception decode(String methodKey, Response response) {
+
+        HttpHeaders responseHeaders = new HttpHeaders();
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/reform/sendletter/CustomFeignErrorDecoder.java]: what is this here for? you don't seem to be using it

File: src/main/java/uk/gov/hmcts/reform/sendletter/CustomFeignErrorDecoder.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        response.headers().entrySet().stream()
+                .forEach(entry -> responseHeaders.put(entry.getKey(), new ArrayList<>(entry.getValue())));
+
+        HttpStatus statusCode = HttpStatus.valueOf(response.status());
+        String statusText = response.reason();
+
+        byte[] responseBody;
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/reform/sendletter/CustomFeignErrorDecoder.java]: this seems unused?

File: src/main/resources/META-INF/spring.factories [hunk 4]
```diff
@@ -1,2 +1,3 @@
 org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
-  uk.gov.hmcts.reform.sendletter.SendLetterAutoConfiguration
+  uk.gov.hmcts.reform.sendletter.SendLetterAutoConfiguration,\
+  uk.gov.hmcts.reform.sendletter.SendLetterConfiguration
```
Comment (timja-kainos) [inline src/main/resources/META-INF/spring.factories]: is this needed?

File: src/main/java/uk/gov/hmcts/reform/sendletter/CustomFeignErrorDecoder.java (hunk 2 above)
Comment (dharmendrak) [reply src/main/java/uk/gov/hmcts/reform/sendletter/CustomFeignErrorDecoder.java]: used now

File: src/main/resources/META-INF/spring.factories (hunk 4 above)
Comment (dharmendrak) [reply src/main/resources/META-INF/spring.factories]: will try without it

Comment (dharmendrak) [reply src/main/resources/META-INF/spring.factories]: removed

File: build.gradle [hunk 5]
```diff
@@ -1,4 +1,5 @@
 plugins {
+    id 'net.ltgt.apt' version '0.14'
```
Comment (timja-kainos) [inline build.gradle]: ?

Comment (dharmendrak) [reply build.gradle]: forgot to remove

Comment (dharmendrak) [reply build.gradle]: removed

**Pull request merged**